### PR TITLE
Use epgsql error record

### PR DESCRIPTION
**Pull request by mworrell**
### Description

Use the epgsql `#error{}` record instead of the `{error, ...}` tuple.

### Checklist

- [ ] documentation updated
- [ ] tests added
- [x] no BC breaks

### CODE REVIEW

File: src/db/z_db_pgsql.erl [hunk 1]
```diff
@@ -181,9 +182,9 @@ connect(Args, RetryCt) ->
                               [Hostname, Port, ?CONNECT_RETRY_SLEEP div 1000, self()]),
                 timer:sleep(?CONNECT_RETRY_SLEEP),
                 connect(Args, RetryCt+10);
-            {error, {error,fatal,<<"53300">>,_ErrorMsg,_ErrorArgs}} ->
+            {error, #error{ codename = too_many_connections }} ->
                 too_many_connections(Args, RetryCt);
-            {error, <<"53300">>} ->
+            {error, too_many_connections} ->
```
Comment (ddeboer) [inline src/db/z_db_pgsql.erl]: Can this one still match?

Comment (mworrell) [reply src/db/z_db_pgsql.erl]: I am actually a  bit unsure.  According to the type specs of the connect not though.

Comment (mworrell) [reply src/db/z_db_pgsql.erl]: Checked again:

```erlang
-type connect_error() ::
        #error{}
      | {unsupported_auth_method, atom()}
      | invalid_authorization_specification
      | invalid_password.
```

Comment (ddeboer) [reply src/db/z_db_pgsql.erl]: In that case, let’s remove the redundant patterns.

Comment (mworrell) [reply src/db/z_db_pgsql.erl]: Let's assume the type spec is valid and adapt our code.

**Pull request merged**